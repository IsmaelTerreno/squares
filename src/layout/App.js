import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import '../styles/App.css';
import StartIntroBlock from '../component/StartIntroBlock';
import SquareGallery from '../container/SquareGallery.container';
import SquareForm from '../container/SquareForm.container';

const App = () => {
  return (
      <Router>
        <div className="container">
            <div className="row justify-content-md-center">
                <div className="col-lg-6 col-md-8 col-sm-12 col-12">
                  <div className="App">
                    <Route path="/" exact={true} render={() =>
                      <StartIntroBlock
                      title="Art with Squares"
                      description="Create customized stylish squares to have your own gallery and show them to the world."
                      buttonLegend="Start to create"
                      buttonUrlAction="/square"
                      />
                    }
                    />
                    <Route path="/square" exact={true} render={() => <SquareForm /> } />
                    <Route path="/square/gallery" exact={true} render={() => <SquareGallery /> }/>
                  </div>
                </div>
            </div>
        </div>
      </Router>
  );
};

export default App;
