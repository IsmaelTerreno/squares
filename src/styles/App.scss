@import "./index.scss";

.App {
  text-align: center;
  padding: 1rem;
}
