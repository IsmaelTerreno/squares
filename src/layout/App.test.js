import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build/index';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme/build/index';
configure({ adapter: new Adapter() });

describe('App', () => {
  test('Should renders appropriately', () => {
    const wrapper = shallow(
      <App/>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
