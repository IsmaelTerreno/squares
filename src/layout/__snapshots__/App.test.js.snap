// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should renders appropriately 1`] = `
<BrowserRouter>
  <div
    className="container"
  >
    <div
      className="row justify-content-md-center"
    >
      <div
        className="col-lg-6 col-md-8 col-sm-12 col-12"
      >
        <div
          className="App"
        >
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/square"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/square/gallery"
            render={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</BrowserRouter>
`;
